Tessera in orion mode & support for Privacy group (#46)
ConsenSys/doc.tessera: docs/Concepts/Privacy-Groups.md

## Legacy Privacy Group

Tessera will automatically create a `legacy` privacy group when a private transaction is sent to a list of recipient public keys using `privateFor`.

When returning private transaction data to a Besu client (as part of a `/receive` response), the `legacy` `privacyGroupId` will also be returned.

The `legacy` `privacyGroupId` is generated by hashing the list of recipient keys. For a given set of recipients, there can be only one `legacy` privacy group.

!!! note
    Besu and Tessera will generate the same `legacy` `privacyGroupId` from the same list of recipients.

An example of a `legacy` privacy group:

```json
{
    "description": "Privacy groups to support the creation of groups by privateFor",
    "members": [
        "98FhPni7u6YspDGcOLl/LgQQwwGPGY4ddm3hmogsqF8=",
        "dzkrEhkHZ/IjHEaQ6teVTU/kMjiwXTI6Ooljcb56w1M="
    ],
    "name": "legacy",
    "privacyGroupId": "jufzisK63xbXDciV0FW1uAi3vXFDNNJpf/M3lUhMiU0=",
    "type": "LEGACY"
}
```

> ## Pantheon Privacy Group

Reviewer: ```suggestion
## Besu Privacy Group
```
No other client uses pantheon as privacy group name. So it would be better to call it Besu privacy group even if the value is pantheon. We could then specify that the value is pantheon because of the previous name of Besu, but that's enough IMO.

Author: I don't think this is correct, we will stick to pantheon privacy group as that is what the group name is created as.

> Tessera supports API methods that enable the creation of `pantheon` privacy groups from a compatible blockchain client (for example Besu).

Reviewer: No other client is compatible with this.
```suggestion
Tessera supports API methods that enable the creation of Besu `pantheon`[^1] privacy groups.

[^1]: Besu was previously named Pantheon and this value is still in use for compatibility
```

Author: for now yes but we want it future proof as there is no reason why Quorum client cannot add support to these methods.

Upon creation, the privacy group is distributed to all members ahead of transaction processing.

When a private transaction contains a `privacyGroupId`, the transaction payload is distributed to all the members of the privacy group.

The `pantheon` `privacyGroupId` is generated by hashing the list of recipient keys *plus a random seed*. This means that for a given set of recipients, there can be many `pantheon` privacy groups.

## Database Storage

A new table called `PRIVACY_GROUP` is added to Tessera's database to store Privacy Group data.

> DDLs will be provided along with release notes to create new privacy group tables.

Reviewer: We only document current features. Si if it's not yet available but will be when the release will be done and this doc released, then you can write using the present tense. A link to the DDLs or to the corresponding doc would be nice.

```suggestion
DDLs are provided with release notes to create new privacy group tables.
```
